### PR TITLE
Add versioning for accounts

### PR DESCRIPTION
**Pull request by aaomidi**

### CODE REVIEW

File: src/sql/azdata.proposed.d.ts [hunk 1]
```diff
@@ -512,4 +512,15 @@ declare module 'azdata' {
...
 	}
 
+	export interface AccountKey {
+		/**
+		 * A version string for an account
+		 */
+		accountVersion?: string;
+	}
+
+	export interface Account {
```
Comment (abist) [inline src/sql/azdata.proposed.d.ts:522]: Is there a reason you're adding these interfaces?

Comment (aaomidi) [reply src/sql/azdata.proposed.d.ts:522]: To add fields to them

Comment (aaomidi) [reply src/sql/azdata.proposed.d.ts:522]: Added some docs explaining what delete does.

Comment (abist) [reply src/sql/azdata.proposed.d.ts:522]: What I meant was I don't see these explicitly used anywhere. Did you want to have your stuff typed to these interfaces?

Comment (aaomidi) [reply src/sql/azdata.proposed.d.ts:522]: https://github.com/microsoft/azuredatastudio/pull/11497/files#diff-6d6a3c51b60a460a2ca1eccde6d7a698R116

https://github.com/microsoft/azuredatastudio/pull/11497/files#diff-6d6a3c51b60a460a2ca1eccde6d7a698R515

Comment (abist) [reply src/sql/azdata.proposed.d.ts:522]: Oh looks like these are generalized for AzureAccount and key. Looks good!

File: extensions/azurecore/src/account-provider/auths/azureAuth.ts [hunk 2]
```diff
@@ -505,7 +511,8 @@ export abstract class AzureAuth implements vscode.Disposable {
 		const account = {
 			key: {
 				providerId: this.metadata.id,
-				accountId: key
+				accountId: key,
+				accountVersion: AzureAuth.ACCOUNT_VERSION,
```
Comment (abist) [inline extensions/azurecore/src/account-provider/auths/azureAuth.ts:515]: wait, key has itself in the field?

Comment (aaomidi) [reply extensions/azurecore/src/account-provider/auths/azureAuth.ts:515]: yeah there's technically the accountKey which contains all the information about the provider, and then the accountId is also technically a key

**Comment by aaomidi** [comment]
@Charles-Gagnon correct.

**Pull request merged**